core/cap-0056.md

```diff
@@ -14,50 +14,49 @@ Protocol version: TBD
 ```
 
 ## Simple Summary
-This CAP proposes an update to `TransactionMeta` along with new host functions to allow for contracts to log messages that can be consumed by downstream systems.
+This CAP proposes an update to `TransactionMeta` along with new host functions
+to allow for contracts to write date to the meta that can be consumed by
+downstream systems.
 
 ## Motivation
-Contract writers will need a way to emit information about what their specific contracts are doing. Downstream consumers of these new messages can also push these to subscribers, allowing users to listen to specific messages.
+Contract writers will need a way to emit information about what their specific
+contracts are doing. Downstream consumers of these new messages can also push
+these to subscribers, allowing users to listen to specific messages.
 
 ### Goals Alignment
 This CAP is aligned with the following Stellar Network Goals:
 * The Stellar Network should make it easy for developers of Stellar projects to create highly usable products
 
 ## Abstract
-This CAP provides host functions for contracts to log messages that will be put
+This CAP provides host functions for contracts to write messages that will be put
 into `TransactionMeta`. It provides a way to cryptographically verify the
 contents of `TransactionMeta`, and also moves `TransactionResult` into the meta
 so they stop taking space in the archives.
 
 ## Specification
 
-### Contract Logs
+### Contract Events
 This CAP makes a change to `TransactionMeta` and `TransactionResult` that will allow
 a user to cryptographically verify anything in `TransactionMetaV3`.
 `TransactionMetaV3` will store separate SHA256 hashes for ledger entry changes and
-contract logs, and the hash of those hashes will be stored in `TransactionResultPairV2`.
+contract events, and the hash of those hashes will be stored in `TransactionResultPairV2`.
 
 How can this be used? The `LedgerHeader` contains the hash of the
 `TransactionResultSetV2`, so the user will first gather all results from the desired
 ledger, hash them, and verify that the hash matches what's in the `LedgerHeader`.
 With `TransactionResultSetV2` verified, the user can then find the `TransactionMetaV3`
-for the transaction they're interested in, hash just contract logs, combine
+for the transaction they're interested in, hash just contract events, combine
 that hash with the other hashes in `TransactionMetaV3` and then verify it against
 the hash stored in `TransactionResultPairV2`.
 
 #### Host functions
 ```rust
-/// Will save val to a list that will be returned to core so the logs can be
-/// put into TransactionMetaV3.logs with type ContractLogType::CONTRACT_INFO.
-func $log(param $val i64) (result i64)
+/// Will save val to a list that will be returned to core so the val can be
+/// put into TransactionMetaV3.events with type ContractEventType::CONTRACT_INFO.
```

Comment:
If we're using the terminology 'events', then '_INFO' feels out of place. It's natural to think of logging levels like info, warn, error, but it is less so to think of events using the same terminology. Instead we could probably refer to the event type as simply `CONTRACT` so that there are contract and system events, in a similar way that in other environments there might be application and system events.

Reply:
Yeah good point. Updated.